lib/localFunction.js

```diff
@@ -48,140 +46,65 @@ LocalFunction.prototype._requestCallBack = function(err, response, body) {
-    if (is.instanceof(val, Buffer) || is.instanceof(val, Uint8Array)) {
-      return {
-        bytesValue: val
-      };
-    }
-    if (isPlainObject(val)) {
-      return {
-        mapValue: {
-          fields: instance._encodeFirestoreValue(val)
-        }
-      };
-    }
-    throw new Error(
-      'Cannot encode ' + val + 'to a Firestore Value.' +
-      ' The emulator does not yet support Firestore document reference values or geo points.'
-    );
-  };
-
-  return _.mapValues(data, encodeHelper);
-};
-
 LocalFunction.prototype._call = function(data, opts) {
   opts = opts || {};
   var operationType;
   var dataPayload;
+
+  var makeFirestoreValue = function(input) {
+    var currentTime = (new Date()).toISOString();
+    return {
+      fields: encodeFirestoreValue(input),
+      createTime: currentTime,
+      updateTime: currentTime
+    };
+  };
+
   if (this.httpsTrigger) {
     this.controller.call(this.name, data || {});
   } else if (this.eventTrigger) {
     if (this._isDatabaseFunc(this.eventTrigger)) {
       operationType = _.last(this.eventTrigger.eventType.split('.'));
-      if (operationType === 'create') {
+      switch (operationType) {
+      case 'create':
         dataPayload = {
           data: null,
           delta: data
         };
-      } else if (operationType === 'update' || operationType === 'write') {
-        dataPayload = {
-          data: data.before,
-          delta: data.after
-        };
-      } else if (operationType === 'delete') {
+        break;
+      case 'delete':
         dataPayload = {
           data: data,
           delta: null
         };
+        break;
+      default: // 'update' or 'write'
+        dataPayload = {
+          data: data.before,
+          delta: data.after
+        };
       }
       opts.resource = this._substituteParams(this.eventTrigger.resource, opts.params);
       this.controller.call(this.name, dataPayload, opts);
     } else if (this._isFirestoreFunc(this.eventTrigger)) {
       operationType = _.last(this.eventTrigger.eventType.split('.'));
-      var currentTime =  (new Date()).toISOString();
-      if (operationType === 'create') {
+      switch (operationType) {
+      case 'create':
         dataPayload = {
-          value: {
-            fields: this._encodeFirestoreValue(data),
-            createTime: currentTime,
-            updateTime: currentTime
-          },
+          value: makeFirestoreValue(data),
           oldValue: {}
         };
-      } else if (operationType === 'update' || operationType === 'write') {
+        break;
+      case 'delete':
         dataPayload = {
-          value: {
-            fields: this._encodeFirestoreValue(data.after),
-            createTime: currentTime,
-            updateTime: currentTime
-          },
-          oldValue: {
-            fields: this._encodeFirestoreValue(data.before),
-            createTime: currentTime,
-            updateTime: currentTime
-          }
+          value: {},
```

Comment:
Just checking: the correct value here is an empty object, not `null`?

Reply:
yep